webapp/src/dogma/features/auth/authSlice.ts

```diff
@@ -16,130 +16,141 @@
-  alert('Cannot sign in Central Dogma web console. Please check your account and password again.');
-  return false;
-});
-
-export const logout = createAsyncThunk('/auth/logout', async () => {
-  const response = await axios.post(`${process.env.NEXT_PUBLIC_HOST}/api/v1/logout`, null, {
-    validateStatus: (status) => status < 500,
-  });
-  if (response.status === HttpStatusCode.Ok) {
-    removeSessionId();
-    return true;
+export const login = createAsyncThunk('/auth/login', async (params: LoginParams, { rejectWithValue }) => {
+  try {
+    const config = {
+      headers: {
+        'Content-Type': 'application/x-www-form-urlencoded',
+      },
+    };
+    const { data } = await axios.post(`${process.env.NEXT_PUBLIC_HOST}/api/v1/login`, params, config);
+    return data;
+  } catch (error) {
+    // TODO(ikhoon): Replace alert with Modal
+    alert('Cannot sign in Central Dogma web console. Please check your account and password again.');
+    if (error.response && error.response.data.message) {
+      return rejectWithValue(error.response.data.message);
+    } else {
+      return rejectWithValue(error.message);
+    }
   }
-  alert('Problem logging out. Please try again.');
-  return false;
 });
 
 interface UserSessionResponse {
   isAuthorized: boolean;
   user?: UserDto;
 }
 
-export const validateSession = createAsyncThunk<UserSessionResponse>('/auth/validate_session', async () => {
-  const response = await axios.get(`${process.env.NEXT_PUBLIC_HOST}/security_enabled`, {
-    validateStatus: (status) => status < 500,
-  });
-  if (response.status === HttpStatusCode.NotFound) {
-    // Anonymous mode
-    removeSessionId();
-    return { isAuthorized: true };
-  }
-
-  if (response.status === HttpStatusCode.Ok) {
-    if (getSessionId() === null) {
-      return { isAuthorized: false };
-    }
-
-    const userResponse = await axios.get(`${process.env.NEXT_PUBLIC_HOST}/api/v0/users/me`, {
-      validateStatus: (status) => status < 500,
-    });
-
-    if (userResponse.status === HttpStatusCode.Ok) {
-      // The current session ID is valid. Login is not required.
-      return { isAuthorized: true, user: userResponse.data as UserDto };
+export const validateSession = createAsyncThunk<UserSessionResponse>(
+  '/auth/validate_session',
+  async (_, { rejectWithValue }) => {
+    try {
+      await axios.get(`${process.env.NEXT_PUBLIC_HOST}/security_enabled`);
+    } catch (error) {
+      if (error.response && error.response.data.message) {
+        return rejectWithValue(error.response.data.message);
+      } else {
+        return rejectWithValue(error.message);
+      }
     }
+  },
+);
 
-    if (userResponse.status === HttpStatusCode.Unauthorized) {
-      removeSessionId();
-      return { isAuthorized: false };
-    }
-  }
-
-  // Should not reach here in the normal case.
-  return { isAuthorized: false };
-});
-
+const sessionId = typeof window !== 'undefined' && localStorage.getItem('sessionId');
```

Comment:
We should not use `sessionId` until it is verified.

Reply:
This one is only for the initial state.

Reply:
Makes sense.